library/RevAPI/Orders.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+namespace RevAPI;
+
+class Orders extends \ArrayIterator
+{
+    /**
+     * @var Rev
+     */
+    protected $rev;
+    
+    protected $data;
+    
+    public function __construct(Rev $rev, $orders_array) {
+        $this->rev = $rev;
+        $this->data = $orders_array;
+        parent::__construct($orders_array['orders']);
+    }
+
+    /**
+     * @return Order
+     */
+    public function current()
+    {
+        $current = parent::current();
+        
+        //Grab the order with the API call to get attachments and comments
+        return $this->rev->getOrder($current['order_number']);
```

Comment:
This is more of an ideology of avoiding side-effects and staying idempotent. That is, when I iterate over this object (however many times), I would not typically expect to get different results. To get different results, I would expect to have to call some other operation on this object that is known to not be idempotent.

Just my 2¢. Wasteful was the wrong word.